packages/block-editor/src/components/inner-blocks/index.js

```diff
@@ -84,31 +99,21 @@ function UncontrolledInnerBlocks( props ) {
 	);
 
 	const classes = classnames( {
-		'has-overlay': enableClickThrough && hasOverlay,
+		'has-overlay': hasOverlay,
 		'is-capturing-toolbar': captureToolbars,
 	} );
 
-	let blockList = (
-		<BlockList
-			{ ...props }
-			ref={ forwardedRef }
-			rootClientId={ clientId }
-			className={ classes }
-		/>
+	const blockList = (
+		<BlockContextProvider value={ context }>
```

Comment:
Is it okay to have `context` `undefined` in some cases? Destructuring could be affected too. I don't have a specific usage in mind though.

Reply:
It's ok to destructure `undefined` in an object. Alternatively we could conditionally add context, but I don't think it matters much. If you conditionally wrap you remount the entire tree when the condition changes, though it's not really a concern here since either a block type provides context or it doesn't.

Reply:
No it's not. It's ok to `spread` undefined like ` { ...undefined }. But this errors: `const { x } = undefined`. 

I don't have a strong opinion on the conditional wrapping but if you keep this always wrapped, probably a default empty object for `context` would be more safe, no? `getBlockContext` returns an object.

Reply:
Right, that's what I said, it's ok to spread, which is the case here:

https://github.com/WordPress/gutenberg/blob/029b1d3ce7ee9b32b49396c1a030b2f22b10b0ae/packages/block-editor/src/components/block-context/index.js#L28

I don't mind if we return an empty object, undefined, or conditionally render it. We can the particular thing